internal/services/firewall/firewall_data_source.go

```diff
@@ -151,6 +152,26 @@ func firewallDataSource() *pluginsdk.Resource {
 				},
 			},
 
+			"autoscale_configuration": {
+				Type:        pluginsdk.TypeSet,
```

Comment:
Please change this to `pluginsdk.TypeList` so that the data source users can reference the nested values.